debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/AbstractReplicationMessageColumn.java

```diff
@@ -62,14 +62,16 @@ public TypeMetadataImpl(String columnName, PostgresType type, String typeWithMod
 
             // TODO: make this more elegant/type-specific
             length = type.getDefaultLength();
+            scale = type.getDefaultScale();
             if (typeModifiers.length > 0) {
                 try {
-                    this.length = Integer.parseInt(typeModifiers[0]);
+                    final String typMod = typeModifiers[0];
+                    this.length = type.length(Integer.parseInt(typMod));
+                    this.scale = type.scale(Integer.parseInt(typMod));
```

Comment:
This looks odd, why do we now get length _and_ scale from the first modifier?

Reply:
I understand it looks odd but for timestamp/time the single modifier affects both scale and length.